src/Microsoft.ML.StandardLearners/Standard/SdcaBinary.cs

```diff
@@ -1666,7 +1680,7 @@ public sealed class Options : BinaryOptionsBase
         }
 
         internal SdcaNonCalibratedBinaryTrainer(IHostEnvironment env, Options options)
-            : base(env, options, options.LossFunction.CreateComponent(env))
+            : base(env, options, options.LossFunction ?? options.LossFunctionFactory.CreateComponent(env))
```

Comment:
The only constructor of `Options` assigns `new LogLoss()` to `LossFunction`. Is it still possible to execute `CreateComponent`?

Reply:
not anymore, LossFunction is null by default

---
In reply to: [262723959](https://github.com/dotnet/machinelearning/pull/2856#discussion_r262723959) [](ancestors = 262723959)

Reply:
Then, user will see a trainer with `null` loss, but that trainer is minimizing `LogLoss`. That's why I implemented a pattern mentioned by Tom.

---
In reply to: [262724703](https://github.com/dotnet/machinelearning/pull/2856#discussion_r262724703) [](ancestors = 262724703,262723959)

Reply:
sure, as per documentation line 1659:
If unspecified, <see cref="LogLoss"/> will be used.

---
In reply to: [262726971](https://github.com/dotnet/machinelearning/pull/2856#discussion_r262726971) [](ancestors = 262726971,262724703,262723959)

Reply:
I might prefer that the value be present (the idea of `null` acting like a particular specific instance is a bit suboptimal), but I don't insist on it I guess...

---
In reply to: [262728446](https://github.com/dotnet/machinelearning/pull/2856#discussion_r262728446) [](ancestors = 262728446,262726971,262724703,262723959)